eth/fetcher/tx_fetcher.go

```diff
@@ -275,6 +284,7 @@ func (f *TxFetcher) Notify(peer string, types []byte, sizes []uint32, hashes []c
 		return nil
 	}
 	announce := &txAnnounce{origin: peer, hashes: unknownHashes, metas: unknownMetas}
+	f.counter++
```

Comment:
This is not thread safe, here we're still running concurrently per peer, the part that makes things sequential is 2 lines down with f.notify <- announce.

That said, we already track the arrival time in waittime, so it seems odd to add a second counter to track the same notion.

Reply:
The blob txs don't get a waittime (in the non-time-travel version) since they skip the waitlist... looks like it's safe to assign one for them anyway and use that to order, though I'd also probably want to rename waittime to arrival time to make it more clear it's independent of the waitlist.  Alternative is to make this an atomic counter.  WDYT?

Reply:
After exploring the code a bit and tried different approaches, I realized that a global-atomic-seq-number is the easiest approach. I would prefer to go with that.